dump-extensions/openpower-dumps/dump_manager_system.cpp

```diff
@@ -112,12 +116,46 @@ sdbusplus::message::object_path
     auto objPath = std::filesystem::path(baseEntryPath) / idString;
     std::time_t timeStamp = std::time(nullptr);
 
+    // Get the generator id from params
+    using InvalidArgument =
+        sdbusplus::xyz::openbmc_project::Common::Error::InvalidArgument;
+    using Argument = xyz::openbmc_project::Common::InvalidArgument;
+    using CreateParameters =
+        sdbusplus::xyz::openbmc_project::Dump::server::Create::CreateParameters;
+
+    std::string generatorId;
+    auto iter = params.find(
+        sdbusplus::xyz::openbmc_project::Dump::server::Create::
+            convertCreateParametersToString(CreateParameters::GeneratorId));
+    if (iter == params.end())
+    {
+        log<level::INFO>(
+            "GeneratorId is not provided. Replacing the string with null");
+    }
+    else
+    {
+        try
+        {
+            generatorId = std::get<std::string>(iter->second);
+        }
+        catch (const std::bad_variant_access& e)
```

Comment:
this needs to be before starting the dump, in the case of an error, error will be returned and the dump will be started

Reply:
here also, as the reply to your comment below:
before creation of dump entry, we do the extraction; handle the cases (gen id given/not) and proceed with dump creation..
Am i missing anything here?

Reply:
Done